File: C++/50. Pow(x, n).cpp
/*To implement pow(x, n), which calculates x raised to the power n (i.e. xn).
 

For example:

Input: x = 2.00000, n = 10
Output: 1024.00000*/

class Solution {
public:
    
    double myPow(double x, int n)
    {
        double c=n;
        return pow(x,c);  // using pow function to get x^n
    }
};

File: Java/50. Pow(x, n).java
// Implement pow(x, n), which calculates x raised to the power n (i.e. xn)
class Solution {
	public double myPow(double x, int n) {
		long N = n;
		
		if(N < 0) {
			x = 1 / x;
			N = -N;
		}

		double answer = 1;
		double product = x;

		for(long i = N; i > 0; i /=2) {
			if( (i % 2 == 1) ){
				answer = answer * product;
			}
			product *= product;
		}

		return answer;
	}
}
